### ISSUE TITLE
Firmware assertion

### ISSUE THREAD

**Issue by dschrempf** [LineageOS/android_device_samsung_gts28vewifi#1]
Hello,

is it necessary to upgrade the firmware to T713ZSU2AQA1?  I have a European device with version  T713XXU2APK6 and do not want to install firmware of Taiwan or Hong Kong, especially because the OS version is the same.  It may make sense to upgrade to T713XXU2BQCO from the United Kingdom which has OS version 7.0 as far as I can see, but please correct me.

Thanks.

**Comment by luk1337** [comment]
Yes, we can't support 7.0 firmware right now due to samsung not releasing 7.0 for SM-T813 and kernel source for it.

**Comment by luca020400** [comment]
We're still waiting for 7.0 kernel drop

**Comment by btv** [comment]
In regards to @dschrempf question, why are you forcing a firmware upgrade?

**Comment by luk1337** [comment]
Updated blobs require updated firmware. BTW soon we'll force everyone to update to 7.0 firmware :3
We are just waiting for Samsung to release T813 image.

**Comment by btv** [comment]
Thank you for the quick response. Would you mind linking me to instructions on how to do this (preferably with heimdal & not odin)?

**Comment by luk1337** [comment]
`heimdall flash --APNHLOS NON-HLOS.bin --TZ tz.mbn --SBL1 sbl1.mbn --RPM rpm.mbn --HYP hyp.mbn --ABOOT aboot.mbn`, that's it.

**Comment by btv** [comment]
Thank you. I'll let you know how it turns out soon-ish.

**Comment by btv** [comment]
Is a full wipe required or recommended? (Same with Dalvik, data, & cache?)

**Comment by luk1337** [comment]
Unless you're on stock ROM you shouldn't have to wipe anything.

**Comment by luk1337** [comment]
Also, if you want to help test out 7.0 stuff, join #titandev @ freenode.

**Comment by btv** [comment]
Awesome. Thank you again.

**Comment by btv** [comment]
Sorry if I'm being annoying. Can you recommend a place to download the firmware from? samsung-firmware.org doesn't seem to be working out very well for me.

**Comment by luk1337** [comment]
@btv oh, I forgot to add `--MODEM modem.bin` to the command. BTW samsung released T813 today, so this / next week we'll update both Lineage and TWRP (doesn't work at all on 7.0 bootloader ;_;).

**Comment by btv** [comment]
@luk1337 Is there another link for T813, waiting 4 hours for the firmware from samsung-firmware is a little frustrating?

**Comment by luk1337** [comment]
Use updato.com like a human being.

**Comment by dschrempf** [comment]
Hello,

does TWRP 3.1.0-1 from April 24 work with the 7.0 firmware?  Is it necessary to flash the firmware before twrp or can I flash from within twrp?

Thanks,
Dominik

**Comment by luk1337** [comment]
Yes, that one will work with 7.0 firmware. I personally wouldn't flash firmware from twrp as it's easier to flash it from heimdall for me lol.

**Comment by dschrempf** [comment]
OK, thank you. It is just hard to find the correct heimdall command line with all those partitions. Yours from above, including the modem flag should work, right? Thanks!

**Comment by luk1337** [comment]
Yes, that's what I used.

**Comment by dschrempf** [comment]
Worked like a charm.  Thank you!
